docs/design/adrs/2024-04-08-use-pull-throughout-for-ingestion.md

```diff
@@ -0,0 +1,47 @@
+# Use Pull approach(Integrated Vectorization) throughout for ingestion
+
+* **Status** - proposed
+* **Proposer:** @komalgrover
+* **Date:** 2024-04-08
+* **Technical Story:** [https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321)
+
+## Context and Problem Statement
+
+There are two ways to ingest data to the search index i.e using Push based approach or a Pull based approach. This repository is currently using Push based approach in which the developer has more control on how the index is created and what values are put into the index.
+We want to add Integrated vectorization (Pull based approach) and give the flexibility to the user to use either of the two approaches. The purpose of this ADR is to document the approach taken to be able to deploy Pull based approach.
+
+## Decision Drivers
+* Ease of deployment
+* Ability to convey the concept of Integrated Vectorization (Pull based approach)
+
+## Considered Options
+* Using Pull approach throughout
+* Using Pull for documents and push for URL embeddings
+
+## Decision outcome
+* Using Pull approach throughout, as it meets all our requirements.
```

Comment:
curious to understand what requirements you considered

Reply:
Requirements are directly related to the decision drivers above. The main purpose of IV in this repository is to demonstrate how IV works. It should be easy to understand & deploy.